pkg/downloader/downloader.go

```diff
@@ -0,0 +1,195 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downloader
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/api"
+	configv2 "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/coordinate"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter"
+	refParam "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter/reference"
+	valueParam "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/parameter/value"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/config/v2/template"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/downloader/dynatraceparser"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/manifest"
+	projectv2 "github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/project/v2"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/log"
+)
+
+//DownloadConfigs get configurations from Dynatrace API and transforms them into objects
+func DownloadConfigs(env manifest.EnvironmentDefinition, listApis map[string]api.Api, projectId string) (projectv2.ConfigsPerApisPerEnvironments, []error) {
+	var errors []error
+	result := make(projectv2.ConfigsPerApisPerEnvironments)
+	result[env.Name] = make(map[string][]configv2.Config)
+
+	token, err := env.GetToken()
+	if err != nil {
+		log.Error("error retrieving token for enviroment %v %v", env.Name, err)
+		errors = append(errors, err)
+		return nil, errors
+	}
+	client, err := rest.NewDynatraceClient(env.Url, token)
+	if err != nil {
+		log.Error("error creating dynatrace client for enviroment %v %v", env.Name, err)
+		errors = append(errors, err)
+		return nil, errors
+	}
+	for _, api := range listApis {
+		creator := dynatraceparser.NewDynatraceParser()
+		configs, errs := createConfigforTargetApi(env, creator, client, api, projectId)
+		//apiconfig, err := createConfigFromApiV2(env, api, projectId, token, client)
+		if len(errs) > 0 {
+			log.Error("error getting configs for env %v %v", env.Name, errs[0])
+			errors = append(errors, errs[0])
+			continue
+		}
+		result[env.Name][api.GetId()] = append(result[env.Name][api.GetId()], configs...)
+	}
+	result[env.Name] = solveDependencies(projectId, result[env.Name])
+	if errors != nil {
+		return nil, errors
+	}
+	return result, nil
+}
+func GetDependencies() map[string][]string {
+	dependenciesPerEnvironment := make(map[string][]string)
+	return dependenciesPerEnvironment
+}
+func solveDependencies(projectId string, configs map[string][]configv2.Config) map[string][]configv2.Config {
```

Comment:
This might be a good case for a parallel foreach implementation https://stackoverflow.com/questions/24238820/parallel-for-loop